cirq/ops/projectors.py

```diff
@@ -0,0 +1,98 @@
+from typing import Any, Dict, Iterable, List, Tuple, TYPE_CHECKING, Union
+
+import numpy as np
+
+from cirq import value
+from cirq.ops import raw_types
+
+if TYPE_CHECKING:
+    import cirq
+
+
+@value.value_equality
+class Projector(raw_types.Gate):
```

Comment:
What is this gate gonna look like in a circuit? It may be a good idea to override `_circuit_diagram_info_`.

Reply:
Done. Also added a test.